src/rust/grapl-web-ui/tests/api/plugin.rs

```diff
@@ -0,0 +1,243 @@
+        );
+
+    let response = app.post("api/plugin/create").multipart(form).send().await?;
+
+    eyre::ensure!(
+        response.status() == actix_web::http::StatusCode::OK,
+        "unexpected response: {:?}",
+        &response
+    );
+
+    let response_body = response.json::<CreateResponse>().await?;
+
+    println!("create response body: {:?}", response_body);
+
+    Ok(response_body)
+}
+
+async fn get_plugin_metadata(
+    app: &TestApp,
+    plugin_id: &uuid::Uuid,
+) -> eyre::Result<GetPluginMetadataResponse> {
+    let response = app
+        .get(format!("api/plugin/get_metadata?plugin_id={plugin_id}").as_str())
+        .send()
+        .await?;
+
+    eyre::ensure!(
+        response.status() == actix_web::http::StatusCode::OK,
+        "unexpected response: {:?}",
+        &response
+    );
+
+    let response_body = response.json::<GetPluginMetadataResponse>().await?;
+
+    println!("metadata response body: {:?}", response_body);
+
+    Ok(response_body)
+}
+
+async fn deploy_plugin(app: &TestApp, plugin_id: &uuid::Uuid) -> eyre::Result<()> {
+    let body = serde_json::json!({
+            "plugin_id": plugin_id,
+    });
+
+    let response = app.post("api/plugin/deploy").json(&body).send().await?;
+
+    eyre::ensure!(
+        response.status() == actix_web::http::StatusCode::OK,
+        "unexpected response: {:?}",
+        &response
+    );
+
+    Ok(())
+}
+
+async fn get_deployment(
+    app: &TestApp,
+    plugin_id: &uuid::Uuid,
+) -> eyre::Result<PluginDeploymentResponse> {
+    let response = app
+        .get(format!("api/plugin/get_deployment?plugin_id={plugin_id}").as_str())
+        .send()
+        .await?;
+
+    eyre::ensure!(
+        response.status() == actix_web::http::StatusCode::OK,
+        "unexpected response: {:?}",
+        &response
+    );
+
+    let response_body = response.json::<PluginDeploymentResponse>().await?;
+
+    println!("response body: {:?}", response_body);
+
+    Ok(response_body)
+}
+
+async fn tear_down(app: &TestApp, plugin_id: &uuid::Uuid) -> eyre::Result<()> {
+    let body = serde_json::json!({
+            "plugin_id": plugin_id,
+    });
+
+    let response = app.post("api/plugin/tear_down").json(&body).send().await?;
+
+    eyre::ensure!(
+        response.status() == actix_web::http::StatusCode::OK,
+        "unexpected response: {:?}",
+        &response
+    );
+
+    Ok(())
+}
+
+async fn get_health(
+    app: &TestApp,
+    plugin_id: &uuid::Uuid,
+) -> eyre::Result<GetPluginHealthResponse> {
+    let response = app
+        .get(format!("api/plugin/get_health?plugin_id={plugin_id}").as_str())
+        .send()
+        .await?;
+
+    eyre::ensure!(
+        response.status() == actix_web::http::StatusCode::OK,
+        "unexpected response: {:?}",
+        &response
+    );
+
+    let response_body = response.json::<GetPluginHealthResponse>().await?;
+
+    println!("response body: {:?}", response_body);
+
+    Ok(response_body)
+}
+
```

Comment:
`// TODO uncomment once analyzers are done`

Reply:
Yeah, I have the TODO on line 25, which calls into this. I thought that was sufficient for TODO. This needed to be commented as well to satisfy dead_code lint, but doesn't seem worth putting a TODO in each place. I'm fine with adding here as well if you prefer, just lemme know.